Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/FillAlgorithm.java

```diff
@@ -89,7 +92,7 @@ private Range generateRangeAndReplaceColor(int row, int col, boolean direction)
 
 		int i;
 		for (i = col; i >= 0; i--) {
-			if (mPixels[row][i] == mReplacementColor) {
+			if (mPixels[row][i] == mReplacementColor || isPixelWithinColorTolerance(mPixels[row][i])) {
```

Comment:
Am 28.05.2016 um 08:16 schrieb Wolfgang Slany:
> Could the `mPixels[row][i] == mReplacementColor || ` be removed here as well as in lines 106 and 122 below, so that this line becomes `if (isPixelWithinColorTolerance(mPixels[row][i])) {` ?

that should be an performance improvement. if it is the exactly the same
color the slower isPixelWithinColorTolerance() function should not be
called.